docs/java-httpcomp/topics/client.md

```diff
@@ -0,0 +1,70 @@
+
+```shell-session
+$ keytool -importkeystore -srckeystore client.p12 -destkeystore java.client.keystore.jks -srcstoretype pkcs12 -alias {{ client_name }}
+```
+
+**Note:** It's safe to ignore the following warning from `keytool`.
+
+```
+The JKS keystore uses a proprietary format. It is recommended to migrate to PKCS12 which is an industry standard format using "keytool -importkeystore -srckeystore client.p12 -destkeystore java.client.keystore.jks -srcstoretype pkcs12".
+```
+
+You'll also need a trust store in JKS format containing the root certificate from your CA. Your Java code will use this trust store to make sure certificates presented by servers were signed by your CA. Create the password and agree to trust your CA certificate (type "yes"). Hold onto the password for this one as well.
+
+```shell-session
+$ keytool -keystore java.client.truststore.jks -alias CARoot -import -file {{ ca_cert }}
+```
+
+Pass your JKS key and trust store into your custom `SSLContext` which you will subsequently use to bootstrap HTTPS clients to make request. All of the following Java code will depend on [org.apache.httpcomponents's httpclient](https://search.maven.org/artifact/org.apache.httpcomponents/httpclient/4.5.10/jar).
+
+```java
+// ...
+KeyStore identityKeyStore = KeyStore.getInstance("jks");
+FileInputStream identityKeyStoreFile = new FileInputStream(new File("java.client.keystore.jks"));
+identityKeyStore.load(identityKeyStoreFile, "<keystore password>".toCharArray());
+
+KeyStore trustKeyStore = KeyStore.getInstance("jks");
+FileInputStream trustKeyStoreFile = new FileInputStream(new File("java.client.truststore.jks"));
+trustKeyStore.load(trustKeyStoreFile, "<truststore password>".toCharArray());
+
+
+SSLContext sslContext = SSLContexts.custom()
+  // load identity keystore
+  .loadKeyMaterial(identityKeyStore, "<keystore password>".toCharArray(), new PrivateKeyStrategy() {
+    @Override
+    public String chooseAlias(Map<String, PrivateKeyDetails> aliases, Socket socket) {
+      return "{{ client_name }}";
+    }
+  })
+  // load trust keystore
+  .loadTrustMaterial(trustKeyStore, null)
+  .build();
+// ...
+```
+
+Now you will be able to issue arbitrary HTTP requests (e.g. GET) over TLS through client authentication to the server in your Java code:
+
+```java
+// ...
+SSLConnectionSocketFactory sslConnectionSocketFactory = new SSLConnectionSocketFactory(sslContext,
+  new String[]{"TLSv1.2", "TLSv1.1"},
+  null,
+  SSLConnectionSocketFactory.getDefaultHostnameVerifier());
+
+CloseableHttpClient client = HttpClients.custom()
+  .setSSLSocketFactory(sslConnectionSocketFactory)
+  .build();
+
+HttpGet _get = new HttpGet("https://server.internal.net:9443/");
```

Comment:
This can be dynamic too— `https://{{ server_name }}:{{ server_port }}`

Then, in onboarding docs, it will sub in the name used in the server cert as well as the port used in whatever server they're connecting to (for example, `443` in nginx).